### ISSUE TITLE
GuildChannel.permissionsFor() is borked

### ISSUE THREAD

**Issue by GregoryGregoryGregory** [hydrabolt/discord.js#1599]
**Please describe the problem you are having in as much detail as possible:**
GuildChannel.permissionsFor() Always returns true

**Include a reproducible code sample here, if possible:**
```js
console.log(`Channels: ${message.guild.channels.size}`);
const yourChannels = message.guild.channels.filter(c => c.permissionsFor(message.member).has("READ_MESSAGES")).size;
console.log(`Your Channels: ${yourChannels}`);
```

**Further details:**

- discord.js version: 12.0.0-dev
- node.js version: 8.0.0
- Operating system: Windows 10
- Priority this issue should have – please be realistic and elaborate if possible:
minor

**Comment by SpaceEEC** [comment]
#1570 should fix this

**Comment by GregoryGregoryGregory** [comment]
@SpaceEEC Ah, thanks!

**Comment by iCrawl** [comment]
Fixed in 00eebd34
